src/FSharp.Stats/Integration/Differentiation.fs

```diff
@@ -26,56 +26,63 @@ module Differentiation =
         let b  = (x1 - x0 - a*t1*t1)/t1
         (2.*a*t) + b
 
-    ///correcture of the Array.iterUntili function; here no boundarys are set to the lower and upper end
-    let private iterUntiliWithoutCutOff (predicate: int -> 'T -> bool) stepSize startIdx (arr: 'T []) =
-        let rec loop  (arr: 'T []) currentIdx =
-            if currentIdx < 0 then None
-            elif currentIdx > arr.Length-1 then None
-            else                                              
-                match predicate currentIdx arr.[currentIdx] with 
-                | true -> Some currentIdx   
-                | _               -> loop arr (currentIdx+stepSize) 
-        loop arr startIdx 
+    module TwoPointDifferentiation =
```

Comment:
Now the only thing i am missing is a description above the module. While you are at it, you can also add one above the differentiation module. Otherwise i have no more to add, the only thing @ZimmerD taking a look at the actual code.